inc/zoo/map/RobinHood.h

```diff
@@ -387,62 +336,89 @@ struct RH_Frontend_WithSkarupkeTail {
             // deadline, index, swarIndex, intraIndex, elementToInsert correct
             // mdp points to the haystack that gave the deadline
-
-            // back this up to pick the element being evicted, if there is one
-            auto mdBackup = *mdp;
-            auto evictedPSL = mdBackup.PSLs().at(intraIndex);
+            auto md = *mdp;
+            auto evictedPSL = md.PSLs().at(intraIndex);
             if(0 == evictedPSL) { // end of eviction chain!
-                assignMetadataElement(
-                    deadline,
-                    elementToInsert.shiftLanesLeft(intraIndex),
-                    mdp
-                );
+                if(SlotCount - 1 <= index) {
+                    throw MaximumProbeSequenceLengthExceeded("full table");
+                }
                 if(0 == relocationsCount) { // direct build of a new value
-                    values_[index].build(std::pair{k, mv}); // inplace?
+                    values_[index].build(
+                        std::piecewise_construct,
+                        std::tuple(k),
+                        std::tuple(mv)
+                    );
+                    *mdp = mdp->blitElement(intraIndex, elementToInsert);
                     return std::pair{values_.data() + index, true};
                 }
                 // the last element is special because it is a
                 // move-construction, not a move-assignment
-                auto fromIndex = relocations[--relocationsCount];
+                --relocationsCount;
+                auto fromIndex = relocations[relocationsCount];
                 values_[index].build(
                     std::move(values_[fromIndex].value())
                 );
+                md_[swarIndex] =
+                    md_[swarIndex].blitElement(
+                        intraIndex, elementToInsert
+                    );
+                elementToInsert = newElements[relocationsCount];
                 index = fromIndex;
+                swarIndex = index / MD::NSlots;
+                intraIndex = index % MD::NSlots;
                 // do the pair relocations
                 while(relocationsCount--) {
                     fromIndex = relocations[relocationsCount];
                     values_[index].value() =
                         std::move(values_[fromIndex].value());
+                    md_[swarIndex] = md_[swarIndex].blitElement(intraIndex, elementToInsert);
+                    elementToInsert = newElements[relocationsCount];
                     index = fromIndex;
+                    swarIndex = index / MD::NSlots;
+                    intraIndex = index % MD::NSlots;
                 }
                 values_[index].value() = std::pair(k, mv);
+                md_[swarIndex] =
+                    md_[swarIndex].blitElement(intraIndex, elementToInsert);
                 return std::pair{values_.data() + index, true};
             }
+            if(HighestSafePSL < evictedPSL) {
```

Comment:
HighestSafePSL = (1 << PSL_Bits) - MD::NSlots - 1;
5 bits 8 slots is 32-9
4 bits 8 slots is 16-9
3 bits 8 slots is 8-9
This seems ok for now, but we'll have to do something explicit to support small PSL in future, probably.

Reply:
Alternately stated: I think this constrains our max load factor heavily.

Reply:
There is something here: for tables so small that the number of lanes in a SWAR has any significance, the whole of a SWAR implementation is counterproductive.
I don't see the argument you make:
For a number of lanes of 8 per SWAR, 8 codes from 2^`PSL_Bits` are unsafe, codes in which you simply can't use SWAR, if we have to test for those, I think the performance would crater